src/MainWindow.cpp

```diff
@@ -0,0 +1,936 @@
+
+    // Organize the buttons into a grid layout
+    createButtonGridLayout();
+
+    // Kick off the start sequence where we check for the existence of the config file
+    attemptConfigFileGeneration();
+}
+
+MainWindow::~MainWindow()
+{
+    // If the bill widget has been created successfully
+    if(m_billWidget != nullptr)
+    {
+        // Delete it and set it to be null
+        delete m_billWidget;
+        m_billWidget = nullptr;
+    }
+
+    // Properly delete the amount available label
+    deleteLabelIfNonNull(m_amountAvailableLabel);
+
+    // Properly delete the amount available line edit
+    deleteLineEditIfNonNull(m_amountAvailableEdit);
+
+    // Properly delete the save button
+    deleteButtonIfNonNull(m_saveButton);
+    deleteButtonIfNonNull(m_addBillButton);
+    deleteButtonIfNonNull(m_deleteBillButton);
+    deleteButtonIfNonNull(m_fundBillButton);
+    deleteButtonIfNonNull(m_defundBillButton);
+
+    // If the bill table widget has been created successfully
+    if(m_billTableWidget != nullptr)
+    {
+        // Delete it and set it to be null
+        delete m_billTableWidget;
+        m_billTableWidget = nullptr;
+    }
+}
+
+void MainWindow::createButtonGridLayout()
+{
+    // Organize the buttons into a grid layout
+    m_buttonGridLayout = new QGridLayout(this);
+    m_buttonGridLayout->addWidget(m_addBillButton, 0, 0, Qt::AlignmentFlag::AlignLeft);
+    m_buttonGridLayout->addWidget(m_deleteBillButton, 0, 1);
+    m_buttonGridLayout->addWidget(m_fundBillButton, 1, 0);
+    m_buttonGridLayout->addWidget(m_defundBillButton, 1, 1);
+    m_buttonGridLayout->addWidget(m_saveButton, 2, 0);
+
+    // Ensure there is sufficient spacing between the buttons so they do not overlap
+    m_buttonGridLayout->setHorizontalSpacing(5);
+
+    // Position the grid layout in the top right corner of the MainWindow
+    m_buttonGridLayout->setGeometry(QRect(250, -5, 250, 100));
+}
+
+void MainWindow::terminateApplication()
+{
+    // Terminate the application
+    QCoreApplication::quit();
+}
+
+void MainWindow::createConfigFileGenerateFailureBox()
+{
+    // Create a message box with appropriate displayed text
+    QMessageBox configFileFailBox;
+    configFileFailBox.setText(m_CONFIG_FILE_GENERATE_FAIL_BOX_PRIMARY_TEXT);
+    configFileFailBox.setInformativeText(m_CONFIG_FILE_GENERATE_FAIL_BOX_INFO_TEXT);
+    configFileFailBox.setStandardButtons(QMessageBox::Ok);
+
+    int configFileFailBoxSelection = configFileFailBox.exec();
+
+    switch(configFileFailBoxSelection)
+    {
+    // If the user presses "Ok", terminate the application
+    case QMessageBox::Button::Ok :
+    {
+        emit conditionToTerminateMet();
+        break;
+    }
+
+    default :
+    {
+        // Terminate the application by default
+        emit conditionToTerminateMet();
+        break;
+    }
+    }
+}
+
+void MainWindow::createInvalidKeyBox()
```

Comment:
Can these QMessageBox creation functions with just an Ok button be turned into a single helper function which just takes in the primary and informative text strings as parameters?

Reply:
Helper method created.